src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -611,8 +612,8 @@ private void processVariableDef(DetailAST ast, DetailAST nextToken) {
      */
     private boolean isViolatingEmptyLineBetweenFieldsPolicy(DetailAST detailAST) {
         return detailAST.getType() != TokenTypes.RCURLY
-                && (!allowNoEmptyLineBetweenFields
-                    || detailAST.getType() != TokenTypes.VARIABLE_DEF);
+                && !(allowNoEmptyLineBetweenFields
+                    && TokenUtil.isOfType(detailAST, TokenTypes.COMMA, TokenTypes.VARIABLE_DEF));
```

Comment:
Can we keep negatiion close to each Expression ?
It is much easier to read code when  negation is as close as possible.

I need vote on this from maintainers.

Reply:
@pbludov , @nmancus1 , @esilkensen , @rdiachenko , please share your thoughts on negation placing.

Reply:
> please share your thoughts on negation placing.

I think that negation of each expression is more clear. in this case.

> If we can give good name .. we can move expression in parentheses in special method.

`isVariableDeclarationList()` ?

Reply:
done, condition is changed, xdoc extended